test/integration/onboarding/wallet-created.test.tsx

```diff
@@ -0,0 +1,117 @@
+import { fireEvent, waitFor } from '@testing-library/react';
+import mockMetaMaskState from '../data/onboarding-completion-route.json';
+import { integrationTestRender } from '../../lib/render-helpers';
+import * as backgroundConnection from '../../../ui/store/background-connection';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../shared/constants/metametrics';
+
+jest.mock('../../../ui/store/background-connection', () => ({
+  ...jest.requireActual('../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+}));
+
+const mockedBakcgroundConnection = backgroundConnection as jest.Mocked<
+  typeof backgroundConnection
+>;
+
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+
+describe('Wallet Created Events', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  afterAll(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  it('are sent when onboarding user who chooses to opt in metrics', async () => {
+    const { getByTestId, getByText } = await integrationTestRender({
+      preloadedState: mockMetaMaskState,
+      backgroundConnection: backgroundConnectionMocked,
+    });
+
+    expect(getByText('Wallet creation successful')).toBeInTheDocument();
+
+    fireEvent.click(getByTestId('onboarding-complete-done'));
+
+    await waitFor(() => {
```

Comment:
As with E2E tests, I imagine these tests will deal with selectors directly and big assertions. So similar to E2E tests, should we encourage the user of local functions and helper classes to abstract these interactions and complex assertions?

Reply:
Not 100% sure we should create those abstractions, at least not from the start.. as the assertions/interactions here are pretty much the same as we would write unit tests.